### ISSUE TITLE
[Bug]: ggt sync swiftly raises CPU temperature to limit

### ISSUE THREAD

**Issue by edlaver** [gadget-inc/ggt#298]
### What happened?

I was experiencing odd machine freezes when developing against Gadget.

I've narrowed this down to my CPU getting thermal throttled as a result of overheating.

I've further narrowed this down to the `ggt sync` command running.

With HWMonitor running (this is on Windows), I can be running at a baseline of around 50-55 degrees (C) on my CPU, with it running at around 1.5-2.0Ghz.

If I run `ggt sync`, temperatures immediately start climbing up to the high 80s and eventually to over 90 degrees (C), with the core running at around 3.8-4.0Ghz.  After a short while, the thermal throttling will kick in and my system will slow to a crawl as it gets slowed down to around 0.2Ghz.

I can replicate this over and over, and its definitely the `ggt sync` command triggering it.

I've tried using the various command line args like `--file-poll-interval 5000` in case that would help, but makes no difference.

I also tried in PowerShell and regular Windows command line, but made no difference.

There seems to be some loop in the code which is maxing out the processor.

Note: Based on the other issue, doesn't seem to be solely a Windows issue, as seems to be happening on Linux also:
https://github.com/gadget-inc/ggt/issues/209

### What did you expect to happen?

`ggt sync` should run at a low level of system resources.

### Steps to reproduce

1. On a Windows machine, install the HWMonitor tool
2. Open HWMonitor and observe the `[CPU name] > Temperatures > Package` value, and the `[CPU name] > Clocks` values
3. In a command prompt, run `ggt sync`
4. Observe the CPU temperature swiftly rise, and the clock speed of the CPU increase dramatically
5. Cancel the `ggt sync` command
6. Observe the CPU temperature fall back to normal, and the clock speed of the CPU decrease to baseline

![image](https://user-images.githubusercontent.com/56447343/222914216-080a0dd3-4d1e-437e-b108-8219fb5a7b5f.png)

### Anything else?

Hopefully there is a simple fix.

I'd also settle for a manual sync option that I can run on demand instead of continuously, e.g. one that runs and does the sync then stops.

I could then run this whenever I need to sync my changes.  As nice as it is to be real time, it's actually not that crucial for my dev workflow.  Running manually would be more like doing a `git` sync on demand.

**Comment by edlaver** [comment]
As a temporary workaround, I've added this in to run a sync on demand.  It will timeout after 15 seconds and kill the process which exits the sync.  This gives enough time to accept the `Merge local files with remote ones` option (or whichever option you want) and process the changes before the process exits.

```json
  "scripts": {
    "sync": "run-p ggt-sync:*",
    "ggt-sync:timeout": "node -e \"setTimeout(() => process.exit(1), 15000);\"",
    "ggt-sync:do-sync": "ggt sync ./gadget/my_gadget_app --app my_gadget_app"
  },
  "devDependencies": {
    "npm-run-all": "^4.1.5"
  }
```

**Comment by scott-rc** [comment]
Hey @edlaver 👋

I just released v0.1.13 which includes #299 that fixes this issue (thanks @airhorns 🎉 )

You can install the latest version via:

```shell
npm install --global @gadgetinc/ggt
```

If the issue persists after upgrading, please re-open this issue.
